src/hotspot/share/gc/shared/cardTable.hpp

```diff
@@ -189,7 +189,7 @@ class CardTable: public CHeapObj<mtGC> {
            "out of bounds access to card marking array. p: " PTR_FORMAT
            " _byte_map: " PTR_FORMAT " _byte_map + _byte_map_size: " PTR_FORMAT,
            p2i(p), p2i(_byte_map), p2i(_byte_map + _byte_map_size));
-    size_t delta = pointer_delta(p, _byte_map_base, sizeof(CardValue));
+    size_t delta = p - _byte_map_base;
```

Comment:
Maybe it's worth a comment that this is specifically NOT using pointer_delta because `_byte_map_base` can be "negative"?  I suspect that situation involves UB, but we've been "getting away" with it so far, so don't try to fix it here.